### ISSUE TITLE
longStackTraceZone not producing meaningful stack traces

### ISSUE THREAD

**Issue by matthewjh** [angular/zone.js#175]
````javascript
var myZone = zone.fork(Zone.longStackTraceZone);

myZone.run(function() {
	function a() {
		setTimeout(function() {
			throw new Error('blahh!');
		});
	}

	a();
});
````
logs:
````
Error: blahh!
at fnNames.map.forEach.global.zone.(anonymous function).arguments.(anonymous function) (http://localhost:8080/node_modules/zone.js/dist/zone.js:350:24)
    at http://localhost:8080/node_modules/zone.js/dist/zone.js:93:23
    at Zone.run (http://localhost:8080/node_modules/zone.js/dist/zone.js:113:17)
    at zoneBoundFn (http://localhost:8080/node_modules/zone.js/dist/zone.js:86:19)
--- Mon Sep 07 2015 13:34:19 GMT+0100 (BST) - 44ms ago
    at _getStacktraceWithUncaughtError (http://localhost:8080/node_modules/zone.js/dist/long-stack-trace-zone.js:39:26)
    at Zone.module.exports.$fork [as fork] (http://localhost:8080/node_modules/zone.js/dist/long-stack-trace-zone.js:97:40)
    at Zone.bind (http://localhost:8080/node_modules/zone.js/dist/zone.js:84:48)
    at Zone.bindOnce (http://localhost:8080/node_modules/zone.js/dist/zone.js:92:17)
    at bindArgumentsOnce (http://localhost:8080/node_modules/zone.js/dist/zone.js:836:29)
    at fnNames.map.forEach.global.zone.(anonymous function) (http://localhost:8080/node_modules/zone.js/dist/zone.js:352:20)
    at fnNames.map.forEach.obj.(anonymous function) (http://localhost:8080/node_modules/zone.js/dist/zone.js:359:37)
    at Zone.run (http://localhost:8080/node_modules/zone.js/dist/zone.js:113:17)
````

This is clearly not useful, as it only includes frames from within zone. The reason for this is that `longStackTraceZone.stackFrameFilter` only filters in frames referring to one of the zone.js files, rather than filtering _out_ these frames which IMO makes more sense.

Inverting the filter produces what I considered to be the expected output:
````javascript
var myZone = zone.fork(Zone.longStackTraceZone);

myZone.stackFramesFilter = function(f) { return !Zone.longStackTraceZone.stackFramesFilter(f); };

myZone.run(function() {
	function a() {
		setTimeout(function() {
			throw new Error('blahh!');
		});
	}

	a();
});
````
````
Error: blahh!
    at http://localhost:8080/index.js:8:10
--- Mon Sep 07 2015 13:37:19 GMT+0100 (BST) - 50ms ago
Error
    at a (http://localhost:8080/index.js:7:3)
    at http://localhost:8080/index.js:12:2
````

@vicb Is this a mistake or is there a rationale for this behavior that I'm just missing…?

**Comment by vicb** [comment]
You're right, that's an error - that I probably introduced as far as I remember :(

**Comment by vicb** [comment]
fixed in #176
